src/http.c

```diff
@@ -150,75 +155,87 @@ int http_receive(
         uint32_t *response_size
 )
 {
-	uint32_t res_size = BUFSZ;
-	char *buffer, *res;
-	int n = 0;
-	int bytes_read = 0;
-	int header_size = 0;
-	int content_size = 0;
+	uint32_t capacity = HTTP_BUFFER_SIZE;
+	char *buffer;
+	uint32_t bytes_read = 0;
+	uint32_t header_size = 0;
+	uint32_t content_size = 0;
 	int chunked = 0;
 
-	buffer = malloc(res_size);
+	buffer = malloc(capacity);
 	if (buffer == NULL)
 		return ERR_HTTP_NO_MEM;
 
-	do {
-		n = safe_ssl_read(tunnel->ssl_handle,
-		                  (uint8_t *) buffer + bytes_read,
-		                  BUFSZ - 1 - bytes_read);
```

Comment:
Did you drop this "- 1" intentionally?

Reply:
Indeed I don't understand why it should be needed - we don't add a terminal `\0` until later on - and when we do we first realloc anyway.

Reply:
maybe this was just a left-over from a previous version of the code that didn't realloc yet.

Reply:
Indeed. Not sure why we always realloc, even if the capacity is sufficient for the final `\0`. Sure, libc may recover a few bytes that might (perhaps) be re-used elsewhere but at the expense of copying the contents of `buffer`. Is it worth it?

Reply:
perhaps just to make sure the final `'\0'` fits in - without having checked if it's necessary at all

Reply:
Sure, but then that's what the "-1" I have removed was about - I think. Why have both? Strange.

Reply:
They have been there since the [initial commit](https://github.com/adrienverge/openfortivpn/blob/171cebae7db0c0dbd5d0609c2d4a52ed98a05182/src/http.c#L71-L81). Maybe the intention was really to shorten the buffer to the needed length.